encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/model/encryptedlogging/EncryptedLog.kt

```diff
@@ -1,80 +1,69 @@
 package com.automattic.encryptedlogging.model.encryptedlogging
 
-import com.yarolegovich.wellsql.core.Identifiable
-import com.yarolegovich.wellsql.core.annotation.Column
-import com.yarolegovich.wellsql.core.annotation.PrimaryKey
-import com.yarolegovich.wellsql.core.annotation.RawConstraints
-import com.yarolegovich.wellsql.core.annotation.Table
-import com.automattic.encryptedlogging.model.encryptedlogging.EncryptedLogUploadState.QUEUED
-import com.automattic.encryptedlogging.utils.DateTimeUtils
+import androidx.room.Entity
+import androidx.room.PrimaryKey
 import java.io.File
-import java.util.Date
 
 /**
  * [EncryptedLog] and [EncryptedLogModel] are tied to each other, any change in one should be reflected in the other.
  * [EncryptedLog] should be used within the app, [EncryptedLogModel] should be used for DB interactions.
  */
 internal data class EncryptedLog(
+    val id: Int = 0,
     val uuid: String,
     val file: File,
-    val dateCreated: Date = Date(),
-    val uploadState: EncryptedLogUploadState = QUEUED,
+    val uploadState: EncryptedLogUploadState = EncryptedLogUploadState.QUEUED,
     val failedCount: Int = 0
 ) {
     companion object {
         fun fromEncryptedLogModel(encryptedLogModel: EncryptedLogModel) =
-                EncryptedLog(
-                        dateCreated = DateTimeUtils.dateUTCFromIso8601(
-                            encryptedLogModel.dateCreated ?: ""
-                        ) ?: Date(),
-                        // Crash if values are missing which shouldn't happen if there are no logic errors
-                        uuid = encryptedLogModel.uuid!!,
-                        file = File(encryptedLogModel.filePath),
-                        uploadState = encryptedLogModel.uploadState,
-                        failedCount = encryptedLogModel.failedCount
-                )
+            EncryptedLog(
+                id = encryptedLogModel.id,
+                // Crash if values are missing which shouldn't happen if there are no logic errors
+                uuid = encryptedLogModel.uuid!!,
+                file = File(encryptedLogModel.filePath),
+                uploadState = encryptedLogModel.uploadState,
+                failedCount = encryptedLogModel.failedCount,
+            )
     }
 }
 
-@Table
-@RawConstraints("UNIQUE(UUID) ON CONFLICT REPLACE")
-internal class EncryptedLogModel(@PrimaryKey @Column private var id: Int = 0) : Identifiable {
-    @Column var uuid: String? = null
-    @Column var filePath: String? = null
-    @Column var dateCreated: String? = null // ISO 8601-formatted date in UTC, e.g. 1955-11-05T14:15:00Z
-    @Column var uploadStateDbValue: Int = QUEUED.value
-    @Column var failedCount: Int = 0
-
-    override fun getId(): Int = id
-
-    override fun setId(id: Int) {
-        this.id = id
-    }
-
+@Entity(
+    tableName = "EncryptedLogEntity",
+)
+internal data class EncryptedLogModel(
```

Comment:
Minor again, but I think we can simply rename `EncryptedLogModel` - I don't do this in Woo during migration as diffs would be huge. 

```suggestion
internal data class EncryptedLogEntity(
```

Reply:
FYI: This is now done: ad976f6fff38fd1f2dc7560cab60ab04f86b1078